prime-router/src/test/kotlin/fhirengine/translation/hl7/FhirTransformerTests.kt

```diff
@@ -0,0 +1,263 @@
+package gov.cdc.prime.router.fhirengine.translation.hl7
+
+import assertk.assertThat
+import assertk.assertions.hasClass
+import assertk.assertions.isEmpty
+import assertk.assertions.isEqualTo
+import assertk.assertions.isFailure
+import assertk.assertions.isFalse
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.fhirTransform.FHIRTransformSchemaElement
+import gov.cdc.prime.router.fhirengine.translation.hl7.schema.fhirTransform.FhirTransformSchema
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.CustomContext
+import gov.cdc.prime.router.fhirengine.translation.hl7.utils.FhirPathUtils
+import org.hl7.fhir.r4.model.Bundle
+import org.hl7.fhir.r4.model.Patient
+import kotlin.test.Test
+
+class FhirTransformerTests {
```

Comment:
I like these tests, but should we add more unit tests to test all the paths in the other worker functions like `setBundleProperty`? Did you run a code coverage report? What does that look like?

Reply:
Ah yea you're probably right, I wrote tests focused on the Acceptance Criteria, but probably should go back and check code coverage and do more true "Unit" tests for the other functions. Will work on that today, hopefully the rest of this is still okay to review.

Reply:
I like the idea of having tests focused on acceptance criteria! I think adding tests for good code coverage is also very important as well, and it helps me better understand the code as well. I don't think code coverage needs be 100% in all cases but these methods are long and have a lot of logic in them that need to have tests.

Reply:
Agreed, yea I think it was mostly laziness and eagerness to get this out for review, but going back and boosting code coverage actually exposed a few issues, so very much worthwhile!

Many of the helper functions were tested already in `FhirToHl7ConverterTests.kt`, so I pulled some of those out into a new file `ConfigSchemaProcessorTests.kt` and added new tests to cover both `getValue` and `getValueAsString`. I was missing some coverage of valueSet cases for `getValue` so I added tests and made some fixes to the function for that. I also was missing some coverage for adding an extension as part of the fhir transform, so I added tests for that, and then made some fixes to that chunk of code too. Code coverage is back to where I want it to be, although I still don't have really isolated unit tests for some functions in `FhirTransfomer`, so I may still go back and add a few there.